src/clients/Deluge.ts

```diff
@@ -237,35 +262,35 @@ export default class Deluge implements TorrentClient {
 	 */
 	private async getTorrentInfo(searchee: Searchee): Promise<TorrentInfo> {
 		try {
-			let torrentLabel: string = undefined;
 			const params = [
 				["state", "progress", "save_path", "label"],
 				{ hash: searchee.infoHash },
 			];
-			const response = await this.call("web.update_ui", params);
-			if (response?.result?.torrents?.[searchee.infoHash] === undefined) {
+			if (!searchee.infoHash)
+				throw new Error("Can't search a torrent without a infoHash");
+
+			const response = (await this.call(
+				"web.update_ui",
+				params
+			)) as InfoResponse;
+			const torrent = response?.result?.torrents?.[searchee.infoHash];
```

Comment:
good variable extraction here

Reply:
yeah still not happy with the amount of nullskips, but at least it's contained.

Reply:
> yeah still not happy with the amount of nullskips, but at least it's contained.

I've almost eliminated them, I believe, in the commit I just made. Take a look.